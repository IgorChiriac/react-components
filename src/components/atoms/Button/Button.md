### Summary

Use `<Button />` whenever you need to render a call to action within a **Zopa interface**.

It comes in a lot of variations to suit different needs.

### Examples

- Primary ( _used for the main call to action, should only appear once per screen._ )

```tsx
import { Button } from '@zopauk/react-components';

<Button styling="primary">Button smash</Button>;
```

- Secondary ( _standard button for most actions_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button styling="secondary">Button smash</Button>;
```

- Disabled state ( _for use when actions are disabled, both primary and secondary_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button disabled={true}>Button smash</Button>;
```

- Text button ( _used to navigate to other pages, always in current window unless external icon is present_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button styling="link">Link smash</Button>;
```

- Warning state ( _for potentially risky actions_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button styling="warning">Proceed</Button>;
```

- Alert state ( _for very dangerous actions_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button styling="alert">Delete</Button>;
```

- Large size ( _rarely to be used, but it's here_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button sizing="large">Big smash</Button>;
```

- Small size ( _mainly used when many actions are available in close proximity_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button sizing="small">Little smash</Button>;
```

- Compact size ( _for when the link style just won't work_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button sizing="compact">Tiny smash</Button>;
```

- With an icon on its right side ( _can be used with arrows, among other icons_ )

```tsx
import { AlertIcon, Button } from '@zopauk/react-components';

<Button rightIcon={<AlertIcon fillColor="#fff" />}>Learn more</Button>;
```

- With an icon on its left side ( _for when the icon conveys the meaning faster than text_ )

```tsx
import { AlertIcon, Button } from '@zopauk/react-components';

<Button leftIcon={<AlertIcon fillColor="#fff" />}>Smash now</Button>;
```

- Full width ( _for when the button should stretch the entire width of the divider_ )

```tsx
import { Button } from '@zopauk/react-components';

<Button fullWidth={true}>Button smash</Button>;
```

- Primary contrast ( _note that the text colour is the same as the background_ )

```tsx { "props": { "style": { "backgroundColor": "#141E64", "border": "2px solid #efefef" } } }
import { Button, colors } from '@zopauk/react-components';

<Button styling="contrastPrimary" contrastColor={colors.neutral.dark}>
  Button smash
</Button>;
```

- Secondary contrast ( _the border should be the colour's shade in 50 or close to that_ )

```tsx { "props": { "style": { "backgroundColor": "#141E64", "border": "2px solid #efefef" } } }
import { Button, colors } from '@zopauk/react-components';

<Button styling="contrastSecondary" contrastColor={colors.neutral.light}>
  Button smash
</Button>;
```

- Text button contrast ( _see above_ )

```tsx { "props": { "style": { "backgroundColor": "#141E64", "border": "2px solid #efefef" } } }
import { Button } from '@zopauk/react-components';

<Button styling="contrastLink">Link smash</Button>;
```

- Button as a custom component

```tsx
import React, { FC } from 'react';
import styled from 'styled-components';
import { buttonStyle } from '@zopauk/react-components';

// could be a gatsby or react-router-dom <Link />
const Link = ({ href, ...rest }) => <a href={href} {...rest} />;

const ButtonLink = styled(Link)`
  ${buttonStyle}
`;

<ButtonLink href="https://www.zopa.com" styling="secondary">
  Zopa
</ButtonLink>;
```
