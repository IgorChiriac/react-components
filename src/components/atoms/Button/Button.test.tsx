import { axe } from 'jest-axe';
import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

it('renders as "primary" and with no a11y violations', async () => {
  const { container } = render(<Button>Primary</Button>);
  expect(container.firstChild).toMatchSnapshot();
  const results = await axe(container.innerHTML);
  expect(results).toHaveNoViolations();
});

it('renders as "secondary"', () => {
  const { container } = render(<Button styling="link">Secondary</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders as "link"', () => {
  const { container } = render(<Button disabled>Secondary</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders wide', () => {
  const { container } = render(<Button fullWidth>Secondary</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders loading as "primary"', () => {
  const { container } = render(<Button loading>Loading</Button>);
  expect(container.firstChild).toMatchSnapshot();
});

it('renders loading as "secondary"', () => {
  const { container } = render(
    <Button styling="secondary" loading>
      Waiting
    </Button>,
  );
  expect(container.firstChild).toMatchSnapshot();
});
