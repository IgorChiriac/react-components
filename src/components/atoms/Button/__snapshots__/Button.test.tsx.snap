// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #EE0505;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #EE0505,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Alert
  </span>
</button>
`;

exports[`Compact button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 14px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 12px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Compact
  </span>
</button>
`;

exports[`Disabled button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
  disabled=""
>
  <span
    class="c1"
  >
    Disabled
  </span>
</button>
`;

exports[`Full width button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Full width
  </span>
</button>
`;

exports[`Large button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 18px;
  line-height: 30px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Large
  </span>
</button>
`;

exports[`Left icon button renders the component with props 1`] = `
.c1 {
  display: block;
  margin-left: 8px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <div>
    My icon
  </div>
  <span
    class="c1"
  >
    Left icon
  </span>
</button>
`;

exports[`Link button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #4A3EDE;
  background-color: transparent;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Link
  </span>
</button>
`;

exports[`Link contrast button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: transparent;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Link contrast
  </span>
</button>
`;

exports[`Primary button renders the component with props with no a11y violations 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Primary
  </span>
</button>
`;

exports[`Primary contrast button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  background-color: #FFFFFF;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #FFFFFF,inset 0 0 0 4px #4A3EDE;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Primary contrast
  </span>
</button>
`;

exports[`Right icon button renders the component with props 1`] = `
.c1 {
  display: block;
  margin-right: 8px;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Right icon
  </span>
  <div>
    My icon
  </div>
</button>
`;

exports[`Secondary button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #4A3EDE;
  background-color: transparent;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE;
}

.c0:hover:enabled {
  opacity: 0.8;
  color: #FFFFFF;
  background-color: #4A3EDE;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 4px #4A3EDE;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Secondary
  </span>
</button>
`;

exports[`Secondary contrast button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: transparent;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: inset 0 0 0 2px #FFFFFF;
}

.c0:hover:enabled {
  opacity: 0.8;
  background-color: transparent;
  box-shadow: inset 0 0 0 2px #FFFFFF;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Secondary contrast
  </span>
</button>
`;

exports[`Small button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 6px 22px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 12px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #FFFFFF;
  background-color: #4A3EDE;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #4A3EDE,inset 0 0 0 4px #FFFFFF;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Small
  </span>
</button>
`;

exports[`Warning button renders the component with props 1`] = `
.c1 {
  display: block;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 30px;
  font-family: "Open Sans",-apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 14px;
  line-height: 24px;
  font-weight: 600;
  cursor: pointer;
  border: none;
  border-radius: 48px;
  color: #0D0A38;
  background-color: #FFB428;
  -webkit-transition: all 140ms ease-in-out;
  transition: all 140ms ease-in-out;
}

.c0:enabled {
  box-shadow: none;
}

.c0:hover:enabled {
  opacity: 0.8;
  box-shadow: none;
}

.c0:active:enabled,
.c0:focus:enabled {
  box-shadow: inset 0 0 0 2px #FFB428,inset 0 0 0 4px #0D0A38;
  outline: none;
}

.c0:active:enabled {
  opacity: 0.4;
}

.c0:disabled {
  cursor: not-allowed;
  color: #C5C6CD;
  background-color: #EAEAEE;
}

<button
  class="c0"
>
  <span
    class="c1"
  >
    Warning
  </span>
</button>
`;
